lib/puppet/http/client.rb

```diff
@@ -150,6 +150,10 @@ def build_resolvers
       resolvers << Puppet::HTTP::Resolver::SRV.new(self, domain: Puppet[:srv_domain])
     end
 
+    if Puppet[:server_list] && !Puppet[:server_list].empty?
+      resolvers << Puppet::HTTP::Resolver::ServerList.new(self, server_list: Puppet[:server_list], default_port: Puppet[:masterport])
+    end
+
     resolvers << Puppet::HTTP::Resolver::Settings.new(self)
     resolvers.freeze
```

Comment:
Okay, I was doing more digging here and it looks like with the ca service in 6.4.x, we always prefer srv first, then `ca_server` if it's set. Even if the first entry in server_list is good, if ca_server is set to an invalid host, it fails to find a good server to connect to. Likewise, if server is set to something valid and server_list is set to something invalid, it will fail and not look at server.

We aren't adhering to that precedent at all with these changes